java-codegen/src/main/resources/org/opensearch/client/codegen/templates/ObjectShape/Equals.mustache

```diff
@@ -0,0 +1,27 @@
+    public boolean equals(Object o) {
+        {{#extendsOtherShape}}
+            if(!super.equals(o)) {
+                return false;
+            }
+        {{/extendsOtherShape}}
+        if(this == o) return true;
+        if(this.getClass() != o.getClass()) return false;
+        {{^hasFieldsToSerialize}}
```

Comment:
This is the wrong conditional to use as it'll ignore query params in request shapes when there's no "body" fields, e.g in `tasks.ListRequest`: https://github.com/opensearch-project/opensearch-java/blob/a7a53e26ec5ad254c4ac325fbcaa84a2ca43bfdc/java-client/src/generated/java/org/opensearch/client/opensearch/tasks/ListRequest.java#L374-L390